packages/next/src/client/components/static-generation-async-storage.external.ts

```diff
@@ -51,6 +51,8 @@ export interface StaticGenerationStore {
 
   isDraftMode?: boolean
   isUnstableNoStore?: boolean
+
+  requestEndedState?: { ended?: boolean }
```

Comment:
Why not `requestEnded: boolean`?

Reply:
Just an object so it's instance can be passed around properly